src/dependencies/DependencyManager.ts

```diff
@@ -77,8 +86,345 @@ export class DependencyManager {
+            if (dependency.version) {
+                return true;
+            } else {
+                return false;
+            }
+        } else if (name === 'Docker for Windows' || name === 'System Requirements') {
+            return dependency.complete;
+        }
+    }
+
+    public async hasPreReqsInstalled(dependencies?: any, optionalInstalled: boolean = false): Promise<boolean> {
+        if (!dependencies) {
+            dependencies = await this.getPreReqVersions();
+        }
+
+        if (!this.isValidDependency(dependencies.node)) {
+            return false;
+        }
+
+        if (!this.isValidDependency(dependencies.npm)) {
+            return false;
+        }
+
+        if (!this.isValidDependency(dependencies.docker)) {
+            return false;
+        }
+
+        if (!this.isValidDependency(dependencies.dockerCompose)) {
+            return false;
+        }
+
+        if (!this.isValidDependency(dependencies.systemRequirements)) {
+            return false;
+        }
+
+        if (process.platform === 'win32') {
+            // Windows
+
+            if (!this.isValidDependency(dependencies.openssl)) {
+                return false;
+            }
+
+            if (!this.isValidDependency(dependencies.buildTools)) {
+                return false;
+            }
+
+            if (!this.isValidDependency(dependencies.dockerForWindows)) {
+                return false;
+            }
+
+        }
+
+        if (process.platform === 'darwin') {
+            // Mac
+            if (!this.isValidDependency(dependencies.xcode)) {
+                return false;
+            }
+        }
+
+        // Optional installs
+        if (optionalInstalled) {
+            if (!this.isValidDependency(dependencies.go)) {
+                return false;
+            }
+
+            if (!this.isValidDependency(dependencies.goExtension)) {
+                return false;
+            }
+
+            if (!this.isValidDependency(dependencies.java)) {
+                return false;
+            }
+
+            if (!this.isValidDependency(dependencies.javaLanguageExtension)) {
+                return false;
+            }
+
+            if (!this.isValidDependency(dependencies.javaDebuggerExtension)) {
+                return false;
+            }
+
+        }
+
+        return true;
+    }
+
+    public async getPreReqVersions(): Promise<any> {
+
+        // Only want to attempt to get extension context when activated.
+        // We store whether the user has confirmed that they have met the System Requirements, so need to access the global state
+
+        if (!context) {
+            context = ExtensionUtil.getExtensionContext();
+            // const _context: vscode.ExtensionContext = ExtensionUtil.getExtensionContext(); // TODO: REMOVE IF THE LINE ABOVE WORKS
```

Comment:
can this be removed?, also i assume if this works you don't need the context in the constructor